odoo/exceptions.py

```diff
@@ -89,6 +91,8 @@ class MissingError(UserError):
 
         When you try to write on a deleted record.
     """
+    # TODO: decide which is best 404 (Not Found) 410 (Gone) 422 (Unproc. Entity)
+    http_status = HTTPStatus.NOT_FOUND  # 404
```

Comment:
410 is better suited in my opinion. 404 is quite generic.

Reply:
ryv's opinion:

> Le problème c'est que c'est un peu des 2 😄 
> Le message le plus courant c'est "Record does not exist or has been deleted.". Après perso je partirais sur le 404, ça me parait plus générique 🙂

And I second this opinion, as doing the following:

```http
POST /web/dataset/call_kw HTTP/1.1
Host: ...
Content-Type: application/json

{
  "jsonrpc": "2.0",
  "method": "call",
  "id": 0,
  "params": {
    "model": "res.partner",
    "method": "write",
    "args": [[100], {"name": "bob"}],
    "kwargs":{}
  }
}
```

Returns a `MissingError` because record 100 doesn't exist.

There's nothing that tells record with id 100 existed before. Hence 410 doesn't apply.

Reply:
Ok for me.